cmd/kubeadm/app/cmd/join.go

```diff
@@ -127,18 +127,21 @@ type joinOptions struct {
 	controlPlane          bool
 	ignorePreflightErrors []string
 	externalcfg           *kubeadmapiv1beta1.JoinConfiguration
+	certificateKey        string
 }
 
 // joinData defines all the runtime information used when running the kubeadm join worklow;
 // this data is shared across all the phases that are included in the workflow.
 type joinData struct {
 	cfg                   *kubeadmapi.JoinConfiguration
 	skipTokenPrint        bool
+	bootstrapClient       clientset.Interface
```

Comment:
AFAIK this is not used by any other phase, so the `bootstrapClient ` can be an internal of `DownloadCerts` (let's keep join as much as possible clean)